templates/rpm/project.spec.erb

```diff
@@ -175,6 +228,23 @@ done
 <%- end -%>
 
 %preun
+# Run pre-uninstall scripts on upgrade if defined
+<%- scripts = get_preremove_actions("upgrade") -%>
```

Comment:
Not sure. We already have comments that refer to nothing in some cases if services aren't delivered in a package.